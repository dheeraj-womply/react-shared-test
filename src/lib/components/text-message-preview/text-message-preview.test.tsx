import React from "react";
import { create as renderer } from "react-test-renderer";
import { render } from "@testing-library/react";
import { TextMessagePreview } from "./index";

describe("Text Message Preview Tests", () => {
  it("renders correctly", () => {
    const component = renderer(
      <TextMessagePreview
        messages={[
          {
            content: (
              <div>
                Fiesta Mexicana requests $160.95 for Dinner - Custom. Pay via{" "}
                <a href="https://pay.xyz/fiestamexicana">
                  https://pay.xyz/fiestamexicana
                </a>
              </div>
            ),
          },
        ]}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with all props", () => {
    const component = renderer(
      <TextMessagePreview
        time="10:41"
        day={"Sunday"}
        messages={[
          {
            content: (
              <div>
                Fiesta Mexicana requests $160.95 for Dinner - Custom. Pay via{" "}
                <a href="https://pay.xyz/fiestamexicana">
                  https://pay.xyz/fiestamexicana
                </a>
              </div>
            ),
          },
        ]}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Text Message Preview Tests with data", () => {
  it("render correctly with data", () => {
    const { rerender, getByTestId } = render(
      <TextMessagePreview
        messages={[
          {
            content: (
              <div>
                Fiesta Mexicana requests $160.95 for Dinner. Pay via{" "}
                <a href="https://pay.xyz/fiestamexicana">
                  https://pay.xyz/fiestamexicana
                </a>
              </div>
            ),
          },
        ]}
      />
    );
    const phoneTime = getByTestId("phone-time");
    expect(phoneTime).toHaveTextContent("9:41");
    const messageTime = getByTestId("message-time");
    expect(messageTime).toHaveTextContent("Tuesday 9:41");
    const message = getByTestId("message-0");
    expect(message).toHaveTextContent(
      "Fiesta Mexicana requests $160.95 for Dinner. Pay via https://pay.xyz/fiestamexicana"
    );
    rerender(
      <TextMessagePreview
        time="10:41"
        day={"Sunday"}
        messages={[
          {
            content: (
              <div>
                Fiesta Mexicana requests $160.95 for Dinner - Custom. Pay via{" "}
                <a href="https://pay.xyz/fiestamexicana">
                  https://pay.xyz/fiestamexicana
                </a>
              </div>
            ),
          },
        ]}
      />
    );
    const customPhoneTime = getByTestId("phone-time");
    expect(customPhoneTime).toHaveTextContent("10:41");
    const customMessageTime = getByTestId("message-time");
    expect(customMessageTime).toHaveTextContent("Sunday 10:41");
    const customMessage = getByTestId("message-0");
    expect(customMessage).toHaveTextContent(
      "Fiesta Mexicana requests $160.95 for Dinner - Custom. Pay via https://pay.xyz/fiestamexicana"
    );
    rerender(
      <TextMessagePreview
        messages={[
          {
            content: (
              <div>
                Fiesta Mexicana requests $160.95 for Dinner. Pay via{" "}
                <a href="https://pay.xyz/fiestamexicana">
                  https://pay.xyz/fiestamexicana
                </a>
              </div>
            ),
          },
          {
            content: "Request Approved",
            direction: "RIGHT",
          },
          {
            content: (
              <div>
                Mark Wood requests $149.61 for Taxi. Pay via{" "}
                <a href="https://pay.xyz/markwood">https://pay.xyz/markwood</a>
              </div>
            ),
          },
        ]}
      />
    );
  });
});
