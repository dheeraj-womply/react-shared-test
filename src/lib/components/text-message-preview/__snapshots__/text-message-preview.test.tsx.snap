// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Message Preview Tests renders correctly 1`] = `
<div
  className="sc-AxjAm hsRpzV"
>
  <div
    className="sc-AxirZ hAYwjx"
  >
    <div
      className="sc-AxiKw bHFHAv"
    >
      <div
        className="sc-AxhCb bzbnNF"
        data-testid="phone-time"
      >
        9:41
      </div>
      <div
        className="sc-AxhUy iltBlB"
      >
        <img
          alt="Signal Wifi Battery Icons"
          src="https://res.cloudinary.com/womply/image/upload/react-shared-components/vitals_TextMsgPreview.svg"
        />
      </div>
    </div>
    <div
      className="sc-AxgMl Oaocg"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#007AFF",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <img
        alt="Womply Logo"
        src="https://res.cloudinary.com/womply/image/upload/react-shared-components/womply_TextMsgPreview.svg"
      />
      <img
        alt="Info Icon"
        src="https://res.cloudinary.com/womply/image/upload/react-shared-components/info_TextMsgPreview.svg"
      />
    </div>
  </div>
  <div
    className="sc-AxheI fQSPAb"
  >
    <div
      className="sc-AxmLO snzpf"
      data-testid="message-time"
    >
      <span
        className="sc-Axmtr drcKvm"
      >
        Tuesday
      </span>
       
      9:41
    </div>
    <div
      className="sc-fzozJi bXYqvA"
      data-testid="message-0"
    >
      <div>
        Fiesta Mexicana requests $160.95 for Dinner - Custom. Pay via
         
        <a
          href="https://pay.xyz/fiestamexicana"
        >
          https://pay.xyz/fiestamexicana
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Text Message Preview Tests renders correctly with all props 1`] = `
<div
  className="sc-AxjAm hsRpzV"
>
  <div
    className="sc-AxirZ hAYwjx"
  >
    <div
      className="sc-AxiKw bHFHAv"
    >
      <div
        className="sc-AxhCb bzbnNF"
        data-testid="phone-time"
      >
        10:41
      </div>
      <div
        className="sc-AxhUy iltBlB"
      >
        <img
          alt="Signal Wifi Battery Icons"
          src="https://res.cloudinary.com/womply/image/upload/react-shared-components/vitals_TextMsgPreview.svg"
        />
      </div>
    </div>
    <div
      className="sc-AxgMl Oaocg"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#007AFF",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <img
        alt="Womply Logo"
        src="https://res.cloudinary.com/womply/image/upload/react-shared-components/womply_TextMsgPreview.svg"
      />
      <img
        alt="Info Icon"
        src="https://res.cloudinary.com/womply/image/upload/react-shared-components/info_TextMsgPreview.svg"
      />
    </div>
  </div>
  <div
    className="sc-AxheI fQSPAb"
  >
    <div
      className="sc-AxmLO snzpf"
      data-testid="message-time"
    >
      <span
        className="sc-Axmtr drcKvm"
      >
        Sunday
      </span>
       
      10:41
    </div>
    <div
      className="sc-fzozJi bXYqvA"
      data-testid="message-0"
    >
      <div>
        Fiesta Mexicana requests $160.95 for Dinner - Custom. Pay via
         
        <a
          href="https://pay.xyz/fiestamexicana"
        >
          https://pay.xyz/fiestamexicana
        </a>
      </div>
    </div>
  </div>
</div>
`;
